### PR TITLE
Doc

### PR DESCRIPTION
**Pull request by STCB**
## Description
Merge de l'update de la DOC dans DEV:
Appliquer au moins ce qui a été fait (80%+ du code doxygéné) pour pouvoir ensuite reprendre sans devoir tout refaire

## Lien tache Jira
https://leomaman.atlassian.net/jira/software/projects/TYPE/boards/4?selectedIssue=TYPE-103

## Changement
Grosse complétion de la docu Doxygen, sur presque tout les .hpp

## Summary by CodeRabbit

New Features:
- Enhanced networking capabilities with the addition of different packet types and management functions, improving data transmission efficiency.
- Introduced new member variables and functions in various components and entities, providing more detailed game data and control.
- Added new server functions for packet handling, client management, and game scene initialization, improving server performance and game flow.

Refactor:
- Updated function signatures and return types in several classes for better consistency and clarity.
- Renamed certain variables for better readability and understanding.

Documentation:
- Added and updated comments and documentation for various classes and functions, improving code readability and maintenance.

Bug Fixes:
- Fixed potential issues with entity movement and collision through new functions and updates in the ECS system.

### CODE REVIEW

File: client/Connection.hpp [hunk 1]
```diff
@@ -44,10 +44,27 @@ namespace RType {
...
+             * @brief The queue of packets to send to the server
+             */
+            std::vector<Network::Packet> sendQueue;
+            /*
+             * @brief The queue of packets received from the server
+             */
+            std::vector<Network::Packet> recvQueue;
+
             void sendPackets();
             void handlePackets();
```
[the remark this answers is not in the document]
Comment (STCB) [reply client/Connection.hpp:69]: It will come later on

File: Network/data/LeaderData.hpp [hunk 2]
```diff
@@ -12,6 +12,10 @@ namespace Network::data {
     {
         struct LeaderData {
             public:
+                /**
+                 * @brief The id of the player that is the leader.
```
Comment (christophechr) [inline Network/data/LeaderData.hpp:16]: "The id of the leader player"

**Review by christophechr — CHANGES_REQUESTED** [review]

File: ecs/AComponent.hpp [hunk 3]
```diff
@@ -38,6 +38,9 @@ namespace ECS {
 
             bool isEnabled = true;
         protected:
+            /** @brief Indicates whether the component is currently active/enabled. */
```
Comment (mangasteak) [inline ecs/AComponent.hpp:41]: il me semble que tu peux rajouter les types

File: server/ServerCore.hpp [hunk 4]
```diff
@@ -31,29 +42,94 @@ namespace ECS {
...
+            /**
+             * @brief The delta time between two frames, also known as the tick time
+             */
             float _deltaTime;
+            /**
+             * @brief The horizontal default speed of the entities
+             */
             float _horizontalSpeed = 500;
+            /**
+             * @brief The vertical default speed of the entities
```
Comment (mangasteak) [inline server/ServerCore.hpp:131]: peut eztre que tu devrais dire la valeur par défaut

**Review by christophechr — APPROVED** [review]

**Review by mangasteak — APPROVED** [review]

**Pull request merged**